src/idl_gen_text.cpp

```diff
@@ -51,11 +51,24 @@ bool Print(T val, Type type, int /*indent*/, Type * /*union_type*/,
            const IDLOptions &opts, std::string *_text) {
   std::string &text = *_text;
   if (type.enum_def && opts.output_enum_identifiers) {
-    auto ev = type.enum_def->ReverseLookup(static_cast<int64_t>(val));
-    if (ev) {
-      text += "\"";
-      text += ev->name;
-      text += "\"";
+    auto enum_values = [&type, &val]() -> std::vector<EnumVal const*> {
```

Comment:
I'm a big fan of lambdas and "linear code", and this trick of using it for local returns is certainly "cute", but this same code with just a vector followed by a few if-thens will actually be easier to understand.

Reply:
Agree. Fixed